Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,254 @@
+  }
+
+  var indexBuffer = [];
+
+  for (var i = 0; i < 3 * raycast.length; i += 3) {
+    indexBuffer[i] = 0;
+    indexBuffer[i + 1] = i / 3 + 1;
+    if (i / 3 + 1 !== raycast.length) indexBuffer[i + 2] = i / 3 + 2;
+    else indexBuffer[i + 2] = 1;
+  }
+
+  this._light.geometry
+    .getBuffer('aVertexPosition')
+    .update(new Float32Array(vertexBuffer));
+
+  this._light.geometry.getIndex().update(new Uint16Array(indexBuffer));
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var centerX = this._object.x;
+  var centerY = this._object.y;
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.getHitBoxes();
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.reduce(function (acc, val) {
+    return acc.concat(val);
+  });
+  var vertices = polygons
+    .map(function (poly) {
+      return poly.vertices;
+    })
+    .reduce(function (acc, val) {
+      return acc.concat(val);
+    });
+
+  function _calculatePOI(angle) {
+    var targetX = centerX + halfOfDiag * Math.cos(angle);
+    var targetY = centerY + halfOfDiag * Math.sin(angle);
+    var minSqDist = halfOfDiag * halfOfDiag;
+    var minPOI = [];
+    for (var poly of polygons) {
+      var poi = gdjs.Polygon.raycastTest(
+        poly,
+        centerX,
+        centerY,
+        targetX,
+        targetY
+      );
+
+      if (poi.collision && poi.closeSqDist <= minSqDist) {
+        minSqDist = poi.closeSqDist;
+        minPOI[0] = poi.closeX;
+        minPOI[1] = poi.closeY;
+      }
+    }
+    if (!minPOI.length) return null;
+    return minPOI;
+  }
+
+  var _raycastResult = [];
+
+  for (var vertex of vertices) {
+    var xdiff = vertex[0] - centerX;
+    var ydiff = vertex[1] - centerY;
+    var angle = Math.atan2(ydiff, xdiff);
+
+    var result = _calculatePOI(angle);
+    if (result) {
+      _raycastResult.push({
+        vertex: result,
+        angle: angle,
+      });
+    }
+
+    // TODO: Check whether we need to raycast these two extra rays or not.
+    var result = _calculatePOI(angle + 0.0001);
+    if (result) {
+      _raycastResult.push({
+        vertex: result,
+        angle: angle + 0.0001,
+      });
+    }
+    var result = _calculatePOI(angle - 0.0001);
+    if (result) {
+      _raycastResult.push({
+        vertex: result,
+        angle: angle - 0.0001,
+      });
+    }
+  }
+
+  return _raycastResult
+    .sort(function (a, b) {
+      if (a.angle < b.angle) return -1;
+      if (a.angle === b.angle) return 0;
+      if (a.angle > b.angle) return 1;
+    })
+    .filter(function (element, index, array) {
+      if (index === 0) return true;
+      if (array[index].angle !== array[index - 1].angle) {
+        return true;
+      }
+      return false;
+    })
+    .map(function (element) {
```

Comment:
filter and map will both create 2 new arrays in memory. See if you can make them in a single loop. 
sort is fine as doing changes in place.